lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -124,6 +122,38 @@ SdfLayerHandle defPrimSpecLayer(const UsdPrim& prim)
 	return defLayer;
 }
 
+bool isTargetLayerHaveOpinion(const UsdPrim& prim)
+{
+    auto editTarget = prim.GetStage()->GetEditTarget();
+    auto layer = editTarget.GetLayer();
+    auto primSpec = layer->GetPrimAtPath(prim.GetPath());
+
+    // to know whether the target layer contains any opinions that 
+    // affect a particular prim, there must be a primSpec for that prim
+    if (!primSpec) {
+        return false;
+    }
+
+    return true;
+}
```

Comment:
To find out if a specific layer contains any opinions that affect a particular prim, we can use the Sdf's GetPrimAtPath() for that prim.